docs/subnets/create-a-vm-blobvm.md

```diff
@@ -0,0 +1,454 @@
+# Create a Blob Virtual Machine (VM)
+
+[toc]
+
+## Introduction
+
+Virtual Machine is like a blueprint for a blockchain and blockchain instantiates from VM, very much similar to a class-object relationship. Issuance of transactions, transaction types, block structure, block building algorithm, keeping pending transactions in the mempool, gossiping transactions to the connected nodes, etc. are defined in a virtual machine.
+
+Virtual Machines can be broken down into components like **Transaction**, **Mempool**, **Network**, **Block**, etc. Each component works together to -
+* Define the representation of the blockchain's state
+* Represent the operations in that state
+* Apply the operations in that state
+
+## BlobVM
+
+In this tutorial, we will learn all about virtual machines and how we can build them by taking references from **[BlobVM](https://github.com/ava-labs/blobvm)**. It's a virtual machine that can be used to instantiate key-value blockchains for storing files like images, videos, etc. in an efficient way.
+
+Blobs are small chunks of data. In BlobVM, we divide a file into small blobs and store them as a key-value pair. The key of these small chunks of a file is linked together as the children of the root. This tree is also stored as JSON data against the root's key.
+
+## How to use a VM
+
+VM modules are served as a binary plugin to the `avalanchego` software. The nodes that want to power their chains using a particular VM, must have the built binary in their avalanchego's `build/plugins` directory. There could be multiple VM plugins in this directory.
```

Comment:
This is dependent on the user's configuration. We can point them to the docs ([ref])(https://docs.avax.network/nodes/maintain/avalanchego-config-flags#build-directory), but this will default to `$GOPATH/src/github.com/ava-labs/avalanchego/build/plugins/` (if they haven't built from source), and `$SOURCE_DIR/build/plugins/` if they are building from source.